admin-dev/themes/new-theme/js/components/contextual-notification.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+/**
+ * This class is responsible for initiating, setting and getting data related to contextual notifications,
+ * that is to say: should we display the notification related to this key identifier in local storage.
+ * It also displays the notification itself
+ */
+
+class ContextualNotification {
+  // all contextual notification data will be stored under this key in local storage
+  private localStorageKey = 'contextual_notifications';
+
+  constructor() {
+    const notificationList = localStorage.getItem(this.localStorageKey);
+
+    // if the contextual_notifications key doesn't exist in localstorage, we set it as empty
+    if (notificationList === null) {
+      localStorage.setItem(this.localStorageKey, JSON.stringify({}));
+    }
+
+    $(document).on('click', '.contextual-notification .close', (Event) => this.disableNotification(Event));
+  }
+
+  private getNotificationList(): Array<boolean> {
+    const notificationList = localStorage.getItem(this.localStorageKey) ?? '{}';
```

Comment:
This seems to be redundant with the constructor which ensures this will never be empty no? We can choose one way or another but both are not needed Maybe the getter solution is better since it really gets the data just in time Meaning nothing will be stored in the local storage unless it is needed